src/main/java/com/github/manolo8/darkbot/config/actions/conditions/TargetTypeCondition.java

```diff
@@ -0,0 +1,100 @@
+package com.github.manolo8.darkbot.config.actions.conditions;
+
+import java.util.Locale;
+
+import org.jetbrains.annotations.NotNull;
+
+import com.github.manolo8.darkbot.Main;
+import com.github.manolo8.darkbot.config.actions.Condition;
+import com.github.manolo8.darkbot.config.actions.Parser;
+import com.github.manolo8.darkbot.config.actions.SyntaxException;
+import com.github.manolo8.darkbot.config.actions.Value;
+import com.github.manolo8.darkbot.config.actions.ValueData;
+import com.github.manolo8.darkbot.config.actions.parser.ParseResult;
+import com.github.manolo8.darkbot.config.actions.parser.ParseUtil;
+import com.github.manolo8.darkbot.config.actions.parser.ValueParser;
+import com.github.manolo8.darkbot.core.entities.Ship;
+
+import eu.darkbot.api.game.entities.Npc;
+import eu.darkbot.api.game.other.EntityInfo.Diplomacy;
+
+@ValueData(name = "is-relationship", description = "Checks the target type", example = "is-relationship(npc, target())")
+public class TargetTypeCondition implements Condition, Parser {
+    public TargetType type;
+    public Value<Ship> ship;
+
+    @Override
+    public @NotNull Condition.Result get(Main main) {
+        Ship target;
+
+        if (type == null) {
+            return Result.ABSTAIN;
+        }
+
+        target = Value.get(ship, main);
+        if (target == null || !target.isValid()) {
+            return type == TargetType.NO_TARGET ? Result.ALLOW : Result.DENY;
+        }
+
+        if (type == TargetType.NPC && target instanceof Npc) {
+            return Result.ALLOW;
+        }
```

Comment:
if the type is npc, it allows, but if it isn't? it doesn't seem to deny, and will probably just abstain.

consider rewriting this into a switch:
```
private boolean matches(Ship target) {
  switch (type) {
    case NO_TARGET: return target == null || !target.isValid();
    case NPC: return target instanceof Npc;
    case ENEMY: return target != null && target.getEntityInfo.isEnemy();
    ---
  }
}
```

then simply return `Result.fromBool(matches(target))`